platform/commonUI/edit/src/actions/SaveAction.js

```diff
@@ -50,23 +67,72 @@ define(
         SaveAction.prototype.perform = function () {
             var domainObject = this.domainObject,
                 $location = this.$location,
-                urlService = this.urlService;
+                urlService = this.urlService,
+                self = this;
+
+            function doWizardSave(domainObject, parent) {
+                var context = domainObject.getCapability("context");
+                var wizard = new CreateWizard(domainObject.useCapability('type'), parent, self.policyService);
+
+                // Create and persist the new object, based on user
+                // input.
+                function persistResult(formValue) {
+                    var parent = wizard.getLocation(formValue),
+                        newModel = wizard.createModel(formValue);
+                    return self.creationService.createObject(newModel, parent);
+                }
+
+                function doNothing() {
+                    // Create cancelled, do nothing
+                    return false;
+                }
+
+                /**
+                 * Add the composees of the 'virtual' object to the
+                 * persisted object
+                 * @param object
+                 * @returns {*}
+                 */
+                function composeObject(object){
+                    return object && self.$q.when(object.hasCapability('composition') && domainObject.hasCapability('composition'))
+                    .then(function(){
+                        return domainObject.useCapability('composition')
+                        .then(function(composees){
+                            return self.$q.all(composees.map(function(composee){
+                                object.getCapability('composition').add(composee);
+                                return object;
+                            })).then(function(){return object});
+                        });
+                    });
+                }
+
+                return self.dialogService.getUserInput(
+                    wizard.getFormStructure(),
+                    wizard.getInitialFormValue()
+                ).then(persistResult, doNothing).then(composeObject);
+            }
 
             // Invoke any save behavior introduced by the editor capability;
             // this is introduced by EditableDomainObject which is
             // used to insulate underlying objects from changes made
             // during editing.
             function doSave() {
-                return domainObject.getCapability("editor").save();
+                //WARNING: HACK
+                //This is a new 'virtual panel' that has not been persisted
+                // yet.
+                if (domainObject.getModel().type === 'telemetry.panel' && !domainObject.getModel().persisted){
```

Comment:
This is an example of where I've had to add specific logic for handling dynamically created panels. In this case, the SaveAction needs a bunch of special logic in order to launch the 'create' dialog which is shown on save when a user has dynamically created a new panel by dragging one plot onto another one.

Not quite sure on the best approach here. I suppose it would be nice to have a completely separate save action for this case, with an appliesTo of: `domainObject.hasCapability('editor') && domainObject.getModel().type === 'telemetry.panel'`

An alternative is to have a single Save action, but with more generic handling of the case where object save requires user input. But how to best encapsulate the logic for determining if user input is required, and how to launch the correct dialog, etc. I imagine there will be other cases in future where we wish to launch a dialog on save.